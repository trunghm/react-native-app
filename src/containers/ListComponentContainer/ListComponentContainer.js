'use strict'

import React from 'react'
import {connect} from 'react-redux'
import ListComponent from './views/index'

// Define which part of the state we're passing to this component
const mapStateToProps = (state) => ({})

// Define the actions this component may dispatch
const mapDispatchToProps = (dispatch) => {
  return {
  }
}

export {ListComponent}

export default connect(mapStateToProps, mapDispatchToProps)(ListComponent)
