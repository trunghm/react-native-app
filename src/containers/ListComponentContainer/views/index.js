import LoginComponent from './ListComponent'
export  default  LoginComponent
