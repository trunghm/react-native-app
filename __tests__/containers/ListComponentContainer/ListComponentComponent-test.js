/**
 * @format
 */

import 'react-native';
import React from 'react';
import {ListComponent} from '../../../src/containers/ListComponentContainer/ListComponentContainer';

// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  ListComponent component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <ListComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <ListComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

});
