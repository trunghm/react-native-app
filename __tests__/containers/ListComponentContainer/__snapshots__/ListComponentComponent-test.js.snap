// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  ListComponent component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  ListComponent component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Back
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "rgba(0,0,0,0.6)",
              "borderBottomWidth": 0.5,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "paddingLeft": 10,
              "paddingRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#39b1ff",
                  "fontSize": 18,
                  "fontWeight": "600",
                }
              }
            >
              Button
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "gray",
                    "borderLeftColor": "transparent",
                    "borderLeftWidth": 10,
                    "borderRightColor": "transparent",
                    "borderRightWidth": 10,
                    "borderTopColor": "gray",
                    "height": 0,
                    "width": 0,
                  },
                  Object {
                    "borderTopWidth": 15,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "rgba(0,0,0,0.6)",
              "borderBottomWidth": 0.5,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "paddingLeft": 10,
              "paddingRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#39b1ff",
                  "fontSize": 18,
                  "fontWeight": "600",
                }
              }
            >
              Avatar
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "gray",
                    "borderLeftColor": "transparent",
                    "borderLeftWidth": 10,
                    "borderRightColor": "transparent",
                    "borderRightWidth": 10,
                    "borderTopColor": "gray",
                    "height": 0,
                    "width": 0,
                  },
                  Object {
                    "borderTopWidth": 15,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "rgba(0,0,0,0.6)",
              "borderBottomWidth": 0.5,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "paddingLeft": 10,
              "paddingRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#39b1ff",
                  "fontSize": 18,
                  "fontWeight": "600",
                }
              }
            >
              GroupButton
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderBottomColor": "gray",
                    "borderLeftColor": "transparent",
                    "borderLeftWidth": 10,
                    "borderRightColor": "transparent",
                    "borderRightWidth": 10,
                    "borderTopColor": "gray",
                    "height": 0,
                    "width": 0,
                  },
                  Object {
                    "borderTopWidth": 15,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
